R/tree_utils.R

```diff
@@ -9,12 +9,11 @@ generate_top_level_map <- function(pc) {
   while ((prev_length <- length(i_not_assigned)) > 0L) { # nolint: implicit_assignment_linter. TODO(#2015): remove this.
     tl_parent[i_not_assigned] <- pc$parent[match(tl_parent[i_not_assigned], pc$id)]
     i_not_assigned <- which(!tl_parent %in% tl_ids)
-    if (length(i_not_assigned) >= prev_length) { # nocov start
-      stop(
-        "Logical error: unassigned set did not shrink. Check file syntax and please report as a lintr bug.",
-        call. = FALSE
-      )
-    } # nocov end
+    # nocov start
+    if (length(i_not_assigned) >= prev_length) {
+      cli_abort("Logical error: unassigned set did not shrink. Check file syntax.")
```

Comment:
Please retain "and report as a bug", we want our logic to catch all such cases & hitting this bug means we have something new to address.

Reply:
Using `cli_abort_internal()` here to retain this.